internal/bminventory/inventory.go

```diff
@@ -4988,7 +4998,7 @@ func (b *bareMetalInventory) UpdateInfraEnvInternal(ctx context.Context, params
 				cluster = nil
 			}
 		}
-		if err = validateArchitectureAndVersion(b.versionsHandler, cluster, infraEnv.CPUArchitecture, infraEnv.OpenshiftVersion); err != nil {
+		if err = validateClusterArchitectureAndVersion(b.versionsHandler, cluster, infraEnv.CPUArchitecture, openshiftVersion); err != nil {
```

Comment:
What if it's a late-binding infraEnv? Would we still want to validate the version with [this function](https://github.com/openshift/assisted-service/pull/6808/files#diff-747a429a0e12307ef65d754c83b6898ce3858b621005e1e7af6e420da1f1657cR4330) or does it not matter if the infraEnv isn't bound to a cluster?

Reply:
I don't think so. IIRC we don't actually require someone to have defined cluster image sets before defining the infraenv. I'd rather leave that validation to the cluster API rather than tie infraenv and cluster concerns more tightly together.

Reply:
Right, we don't require a cluster image set for InfraEnv creation. That makes sense. Thanks!